### PR TITLE
fix error of setting  negative value for containerLogMaxSize

### LINKED ISSUE #99650: kubelet: setting a negative value in containerLogMaxSize raises an error
Setting a negative value in `containerLogMaxSize` currently makes kubelet crash with a stack trace.

The [`parseMaxSize` function](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/logs/container_log_manager.go#L132) function raises an error when the value is negative, but the comment on [`LogRotatePolicy.MaxSize`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/logs/container_log_manager.go#L69) says that setting a negative value disables log rotation.

At the moment there’s no way to disable log rotation and I believe removing lines 141 to 143 from `parseMaxSize`, would allow this.

### PR DESCRIPTION
**Pull request by CaoDonghui123**
#### What type of PR is this?

/kind bug
#### Which issue(s) this PR fixes:

Fixes #99650

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by wangyysde** [comment]
I have verified the #99650.  The kubelet will started normally  when the negative value quoted by double quotation.

File: pkg/kubelet/logs/container_log_manager.go [hunk 1]
```diff
@@ -138,9 +138,6 @@ func parseMaxSize(size string) (int64, error) {
 	if !ok {
 		return 0, fmt.Errorf("invalid max log size")
 	}
-	if maxSize < 0 {
-		return 0, fmt.Errorf("negative max log size %d", maxSize)
-	}
```
Comment (wzshiming) [inline pkg/kubelet/logs/container_log_manager.go:143]: https://github.com/kubernetes/kubernetes/blob/d937ba268a8ed814db74ee6b28f4cc7d4ea9e461/pkg/kubelet/logs/container_log_manager.go#L250

This value will be used here, and it is not allowed to be negative.

**Review by wzshiming — CHANGES_REQUESTED** [review]

File: pkg/kubelet/logs/container_log_manager.go (hunk 1 above)
Comment (CaoDonghui123) [reply pkg/kubelet/logs/container_log_manager.go:143]: I notice this value, it is ok for the negative value, and  I don't see where it's not allowed

Comment (wzshiming) [reply pkg/kubelet/logs/container_log_manager.go:143]: https://github.com/kubernetes/kubernetes/blob/d937ba268a8ed814db74ee6b28f4cc7d4ea9e461/pkg/kubelet/logs/container_log_manager.go#L64-L73
> Negative number means to disable container log rotation. 

If it is a negative number, it is actually not necessary to start this. maybe skip better.
If it is 0, is it using a default?

https://github.com/kubernetes/kubernetes/blob/d937ba268a8ed814db74ee6b28f4cc7d4ea9e461/pkg/kubelet/logs/container_log_manager.go#L174-L182

``` golang
// Start the container log manager. 
 func (c *containerLogManager) Start() { 
	if c.policy.MaxSize < 0 {
		return 
	}
 	// Start a goroutine periodically does container log rotation. 
 	go wait.Forever(func() { 
 		if err := c.rotateLogs(); err != nil { 
 			klog.Errorf("Failed to rotate container logs: %v", err) 
 		} 
 	}, logMonitorPeriod) 
 } 
```

Comment (CaoDonghui123) [reply pkg/kubelet/logs/container_log_manager.go:143]: I get it. I will try to improve it

**Comment by CaoDonghui123** [comment]
ping @ehashman @andrewsykim

**Comment by CaoDonghui123** [comment]
ping @wzshiming

**Comment by wzshiming** [comment]
@CaoDonghui123 We are in code-freeze for 1.21

**Comment by CaoDonghui123** [comment]
ping @ehashman

**Review by ehashman — COMMENTED** [review]
/lgtm

**Comment by finnickkim** [comment]
-

**Comment by CaoDonghui123** [comment]
ping @mrunalp @klueska

File: pkg/kubelet/logs/container_log_manager.go [hunk 2]
```diff
@@ -161,6 +158,10 @@ func NewContainerLogManager(runtimeService internalapi.RuntimeService, osInterfa
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse container log max size %q: %v", maxSize, err)
 	}
+	//Negative number means to disable container log rotation
```
Comment (mrunalp) [inline pkg/kubelet/logs/container_log_manager.go:161]: nit: Add space after //

**Review by ehashman — COMMENTED** [review]
/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**